### PR TITLE
Remove compatibility shim for bytes type

### PR DESCRIPTION
**Pull request by jdufresne**
## Summary of changes

The type `bytes` is available on all supported Pythons. Makes the code more forward compatible with Python 3.

### Pull Request Checklist
- [x] News fragment added in changelog.d. See [documentation](http://setuptools.readthedocs.io/en/latest/developer-guide.html#making-a-pull-request) for details

### CODE REVIEW

File: setuptools/glob.py [hunk 1]
```diff
@@ -92,8 +90,8 @@ def _iglob(pathname, recursive):
 
 def glob1(dirname, pattern):
     if not dirname:
-        if isinstance(pattern, binary_type):
-            dirname = os.curdir.encode('ASCII')
+        if isinstance(pattern, bytes):
+            dirname = os.curdir.decode('ASCII')
```
Comment (jaraco) [inline setuptools/glob.py]: So while I think you're right about this change (encode->decode), it should probably be in a separate commit and maybe even a separate PR, as it's very different from the stated intention of the commit.

File: setuptools/glob.py [hunk 2]
```diff
@@ -129,8 +127,8 @@ def glob2(dirname, pattern):
 # Recursively yields relative pathnames inside a literal directory.
 def _rlistdir(dirname):
     if not dirname:
-        if isinstance(dirname, binary_type):
-            dirname = binary_type(os.curdir, 'ASCII')
+        if isinstance(dirname, bytes):
+            dirname = os.curdir.decode('ASCII')
```
Comment (jaraco) [inline setuptools/glob.py]: Again, this seems like a change to do more than just replace binary_type with bytes. How did this code work before?

File: setuptools/glob.py (hunk 1 above)
Comment (benoit-pierre) [reply setuptools/glob.py]: I think this is wrong: the goal of `glob` is to support this usage too:
```python
python -c 'from setuptools.glob import glob; print(glob(b"*.rst"))'
```
no?
in any case, this fails with this PR.

Comment (benoit-pierre) [reply setuptools/glob.py]: On `master`, from setuptools directory, the results are:
```python
[b'README.rst', b'towncrier_template.rst', b'CHANGES.rst']
```

Comment (benoit-pierre) [reply setuptools/glob.py]: This is probably a good time to add some tests.

Comment (benoit-pierre) [reply setuptools/glob.py]: It looks like the intent of the patch was to change to:
```python
dirname = os.curdir.encode('ASCII')
```

**Comment by jdufresne** [comment]
> This is probably a good time to add some tests.

There is no behavioral change. It is a pure internal refactoring. Do you still want to see tests for this PR? Let me know. If so, I'll add them.

**Comment by benoit-pierre** [comment]
> Do you still want to see tests for this PR? Let me know. If so, I'll add them.

Yes, here are some simple ones to prevent future regressions:
```python
import pytest

from setuptools.glob import glob

from .files import build_files


@pytest.mark.parametrize('tree, pattern, matches', (
    ('', b'', []),
    ('', '', []),
    ('''
     appveyor.yml
     CHANGES.rst
     LICENSE
     MANIFEST.in
     pyproject.toml
     README.rst
     setup.cfg
     setup.py
     ''', '*.rst', ('CHANGES.rst', 'README.rst')),
    ('''
     appveyor.yml
     CHANGES.rst
     LICENSE
     MANIFEST.in
     pyproject.toml
     README.rst
     setup.cfg
     setup.py
     ''', b'*.rst', (b'CHANGES.rst', b'README.rst')),
))
def test_glob(monkeypatch, tmpdir, tree, pattern, matches):
    monkeypatch.chdir(tmpdir)
    build_files({name: '' for name in tree.split()})
    assert list(sorted(glob(pattern))) == list(sorted(matches))
```
To be put in `setuptools/tests/test_glob.py`.

**Comment by jdufresne** [comment]
Thanks. I've added the suggested tests (which do fail with the mistake I temporarily had).
